test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -676,93 +733,149 @@ UCS_TEST_P(test_uct_cm_sockaddr, cm_server_reject)
 
 UCS_TEST_P(test_uct_cm_sockaddr, many_clients_to_one_server)
 {
-    int num_clients = ucs_max(2, 100 / ucs::test_time_multiplier());;
+    int i, no_disconnect_eps_cnt = 0;
     entity *client_test;
+    time_t seed = time(0);
+    ucs_time_t deadline;
+
+    m_is_enhance_test = true;
 
     /* Listen */
     cm_start_listen();
 
     /* Connect */
     /* multiple clients, each on a cm of its own, connecting to the same server */
-    for (int i = 0; i < num_clients; ++i) {
+    for (i = 0; i < m_enhance_clients_num; ++i) {
         client_test = uct_test::create_entity();
         m_entities.push_back(client_test);
+
         client_test->max_conn_priv = client_test->cm_attr().max_conn_priv;
         client_test->connect(0, *m_server, 0, m_connect_addr, client_cm_priv_data_cb,
                              client_connect_cb, client_disconnect_cb, this);
     }
 
     /* wait for the server to connect to all the clients */
     wait_for_client_server_counters(&m_server_connect_cb_cnt,
-                                    &m_client_connect_cb_cnt, num_clients);
+                                    &m_client_connect_cb_cnt, m_enhance_clients_num);
 
-    EXPECT_EQ(num_clients, m_server_recv_req_cnt);
-    EXPECT_EQ(num_clients, m_client_connect_cb_cnt);
-    EXPECT_EQ(num_clients, m_server_connect_cb_cnt);
-    EXPECT_EQ(num_clients, (int)m_server->num_eps());
+    EXPECT_EQ(m_enhance_clients_num, m_server_recv_req_cnt);
+    EXPECT_EQ(m_enhance_clients_num, m_client_connect_cb_cnt);
+    EXPECT_EQ(m_enhance_clients_num, m_server_connect_cb_cnt);
+    EXPECT_EQ(m_enhance_clients_num, (int)m_server->num_eps());
 
     /* Disconnect */
-    for (int i = 0; i < num_clients; ++i) {
+    srand(seed);
+    UCS_TEST_MESSAGE << "Using seed: " << seed;
+    pthread_mutex_init(&m_lock, NULL);
+    m_all_eps = (ep_state_t*)calloc(2 * m_enhance_clients_num, sizeof(ep_state_t));
```

Comment:
use `std::vector<ep_state_t>`

Reply:
why is this better here? an array makes it easier to insert elements in the middle as well...

Reply:
vector allows to initialize by index. can use std::vector::resize(n, value) then operator [] and =